File: src/com/mojang/escape/level/block/LootBlock.java
package com.mojang.escape.level.block;

import com.mojang.escape.*;
import com.mojang.escape.entities.*;
import com.mojang.escape.gui.BasicSprite;
import com.mojang.escape.gui.Sprite;

public class LootBlock extends Block {
	private boolean taken = false;
	private Sprite sprite;

	public LootBlock() {
		sprite = new BasicSprite(0, 0, 0, 16 + 2, Art.INSTANCE.getCol(0xffff80));
		addSprite(sprite);
		setBlocksMotion(true);
	}

	public void addEntity(Entity entity) {
		super.addEntity(entity);
		if (!taken && entity instanceof Player) {
			sprite.setRemoved(true);
			taken = true;
			setBlocksMotion(false);
			((Player) entity).loot++;
			Sound.Companion.getPickup().play();
			
		}
	}

	public boolean blocks(Entity entity) {
		if (entity instanceof Player) return false;
		return getBlocksMotion();
	}
}

File: src/com/mojang/escape/level/block/LootBlock.kt
package com.mojang.escape.level.block

import com.mojang.escape.Art
import com.mojang.escape.Sound
import com.mojang.escape.entities.Entity
import com.mojang.escape.entities.Player
import com.mojang.escape.gui.BasicSprite

class LootBlock: Block() {
    private var taken = false
    private val sprite = BasicSprite(0.0, 0.0, 0.0, 16 + 2, Art.getCol(0xFFFF80))

    init {
        addSprite(sprite)
        blocksMotion = true
    }

    override fun addEntity(entity: Entity) {
        super.addEntity(entity)
        if (!taken && entity is Player) {
            sprite.removed = true
            taken = true
            blocksMotion = false
            entity.loot++
            Sound.pickup.play()
        }
    }

    override fun blocks(entity: Entity): Boolean {
        if (entity is Player) return false
        return blocksMotion
    }
}